packages/ag-charts-community/src/scene/transformable.ts

```diff
@@ -0,0 +1,127 @@
+import { Matrix } from './matrix';
+import { Node, SceneChangeDetection } from './node';
+
+type Constructor<T> = new (...args: any[]) => T;
+
+type MatrixTransformType<T> = T & {
+    updateMatrix(matrix: Matrix): void;
+};
+
+export function MatrixTransform<N extends Node>(Parent: Constructor<N>) {
+    const ParentNode = Parent as Constructor<Node>;
+
+    // Make sure we don't mixin `MatrixTransformInternal` multiple times.
+    let nextType = Parent;
+    while (nextType != null) {
+        if (nextType.name === 'MatrixTransformInternal') {
+            return Parent as unknown as Constructor<MatrixTransformType<N>>;
+        }
+        nextType = Object.getPrototypeOf(nextType);
+    }
+
+    const TRANSFORM_MATRIX = Symbol('matrix_transform');
+    class MatrixTransformInternal extends ParentNode {
+        private [TRANSFORM_MATRIX] = new Matrix();
+
+        private _dirtyTransform = false;
+        override markDirtyTransform() {
+            this._dirtyTransform = true;
+            super.markDirtyTransform();
+        }
+
+        protected updateMatrix(_matrix: Matrix) {
+            // For override by sub-classes.
```

Comment:
nit: if it needs to be defined then shouldn't it be an `abstract`?

Reply:
I tried to make this `abstract`, but it looks like a limitation of the mixin approach that you can't, I end up with a runtime error of:
```
TypeError: (intermediate value).updateMatrix is not a function
```